widget/calendar.go

```diff
@@ -0,0 +1,229 @@
+func (g *calendarLayout) MinSize(objects []fyne.CanvasObject) fyne.Size {
+	rows := g.countRows(objects)
+	return fyne.NewSize((float32(g.cellSize)+g.padding)*7, (float32(g.cellSize)+g.padding)*float32(rows))
+}
+
+// Calendar creates a new date time picker which returns a time object
+type Calendar struct {
+	widget.BaseWidget
+	canvas       fyne.Canvas
+	calendarTime time.Time
+
+	monthPrevious *widget.Button
+	monthNext     *widget.Button
+	monthLabel    *widget.RichText
+
+	day   int
+	month int
+	year  int
+
+	dates *fyne.Container
+
+	onSelected func(time.Time)
+
+	cellSize float32
+	padding  float32
+}
+
+func (c *Calendar) daysOfMonth() []fyne.CanvasObject {
+	start, _ := time.Parse("2006-1-2", strconv.Itoa(c.year)+"-"+strconv.Itoa(c.month)+"-"+strconv.Itoa(1))
+
+	buttons := []fyne.CanvasObject{}
+
+	//account for Go time pkg starting on sunday at index 0
+	dayIndex := int(start.Weekday())
+	if dayIndex == 0 {
+		dayIndex += 7
+	}
+
+	//add spacers if week doesn't start on Monday
+	for i := 0; i < dayIndex-1; i++ {
+		buttons = append(buttons, layout.NewSpacer())
+	}
+
+	for d := start; d.Month() == start.Month(); d = d.AddDate(0, 0, 1) {
+
+		dayNum := d.Day()
+		s := strconv.Itoa(dayNum)
+		var b fyne.CanvasObject = widget.NewButton(s, func() {
+
+			selectedDate := c.dateForButton(dayNum)
+
+			c.onSelected(selectedDate)
+
+			c.hideOverlay()
+		})
+
+		buttons = append(buttons, b)
+	}
+
+	return buttons
+}
+
+func (c *Calendar) dateForButton(dayNum int) time.Time {
+	oldName, off := c.calendarTime.Zone()
+	return time.Date(c.year, time.Month(c.month), dayNum, c.calendarTime.Hour(), c.calendarTime.Minute(), 0, 0, time.FixedZone(oldName, off)).In(c.calendarTime.Location())
+}
+
+func (c *Calendar) hideOverlay() {
+	overlayList := c.canvas.Overlays().List()
+	overlayList[0].Hide()
+}
+
+func (c *Calendar) monthYear() string {
+	return time.Month(c.month).String() + " " + strconv.Itoa(c.year)
+}
+
+func (c *Calendar) calendarObjects() []fyne.CanvasObject {
+	textSize := float32(8)
+	columnHeadings := []fyne.CanvasObject{}
+	for i := 0; i < 7; i++ {
+		j := i + 1
+		if j == 7 {
+			j = 0
+		}
+
+		var canvasObject fyne.CanvasObject = canvas.NewText(strings.ToUpper(time.Weekday(j).String()[:3]), color.NRGBA{0xFF, 0xFF, 0xFF, 0xBF})
+		canvasObject.(*canvas.Text).TextSize = textSize
+		canvasObject.(*canvas.Text).Alignment = fyne.TextAlignCenter
+		columnHeadings = append(columnHeadings, canvasObject)
+	}
+	columnHeadings = append(columnHeadings, c.daysOfMonth()...)
+
+	return columnHeadings
+}
+
+// CreateRenderer returns a new WidgetRenderer for this widget.
+// This should not be called by regular code, it is used internally to render a widget.
+func (c *Calendar) CreateRenderer() fyne.WidgetRenderer {
+	c.monthPrevious = widget.NewButtonWithIcon("", theme.NavigateBackIcon(), func() {
+		c.month--
+		if c.month < 1 {
+			c.month = 12
+			c.year--
+		}
+		c.monthLabel.ParseMarkdown(c.monthYear())
```

Comment:
Why are you parsing markdown here and everywhere else? It seems to me like it work just fine without it in a regular `widget.Label`.

Reply:
Good question, a hangover from the app this widget was initially designed for, working well with `widget.Label` instead